andrepereiradasilva/joomla-cms
# Associations edit

Pull Request for New Issue.
### Summary of Changes

In a multilingual install with associations, when you change the item language the associations will be wrong.

This PR makes the associations be reseted on language changes.
Also:
- allows to add associations on item create (that was disabled because of this problem)
- allows to change the language on item edit (that was disabled because of this problem)
### Testing Instructions
1. Use latest 3.7.x, multilingual with associations enabled
2. Apply patch 
3. For articles, categories, contacts, menus and newsfeeds check:
4. Adding associations when creating an item (try changing the languages)
5. Changing languages when editng an item
### Documentation Changes Required

None.
### Notes

I introduced here two new methods: `JLanguageHelper::getContentLanguages` (get the content languages from `#__languages`) and `JLanguageHelper::getInstalledLanguages` (get the installed languages from `#__extensions` table).

Why this two new methods? 
- because `JLanguageHelper::getLanguages()` always return the published languages ...
- JLanguageHelper::getLanguages() **does not check if the content languages are installed** which, of course, can, and will, lead to errors if you, for instance, unintall a language package.
- because `JLanguageHelper::getLanguages()` is not IMHO semantic correct: get what languages? installed? content languages?
- among more things (like performance)

This two methods will allow to do all languages related stuff across the core in the future (replacing several hardcoded queries, etc).
